### PR TITLE
Populate stores using inline entities declared in the manifest

### PR DESCRIPTION
**Pull request by mykmartin**
This also uncovered a bunch of errors in the existing entity/reference code (particularly with `dataClone`).

### CODE REVIEW

File: src/runtime/manifest.ts [hunk 1]
```diff
@@ -1257,45 +1256,28 @@ ${e.message}
...
-        name,
-        id,
-        storageKey: item.source,
-        tags,
-        originalId,
-        claims,
-        description: item.description,
-        version: item.version,
-        origin: item.origin,
+        type, name, id, storageKey: item.source, tags,
```
Comment (Cypher1) [inline src/runtime/manifest.ts:1259]: Personally I find this harder to read but that's a non-blocking opinion, just feedback in case this becomes common

Comment (mykmartin) [reply src/runtime/manifest.ts:1259]: I agree that the expanded view is a bit easier, but I think with all but one of the args being keyword only, saving the vertical space is worth it (although I originally collapsed this when I was introducing another call to `newStore` for the inline case - so there'd be 3 calls in the function - but that ended up re-using the existing one at the end of the function).

**Comment by Cypher1** [comment]
@mykmartin did you work out what the failure was?

**Comment by mykmartin** [comment]
One of the flakey ones: SameHandleManagerTest

**Comment by Cypher1** [comment]
All good then. Thanks
Joshua Pratt | Arcs | jopra@google.com

**Pull request merged**